lib/registerTypes/RegisterTypes.h

```diff
@@ -208,9 +209,9 @@ void registerTypesMapObjects2(Serializer &s)
 	s.template registerType<CBonusSystemNode, CArtifactInstance>();
 	s.template registerType<CArtifactInstance, CCombinedArtifactInstance>();
 
-	//s.template registerType<CObstacleInstance>();
-		s.template registerType<CObstacleInstance, MoatObstacle>();
-		s.template registerType<CObstacleInstance, SpellCreatedObstacle>();
+	s.template registerType<Obstacle, StaticObstacle>();

```

Comment:
This is a problem. You registering 4 classes where were 3. This completely breaks saves compatibilty.